pages/glossary/visible-inner-text.md

```diff
@@ -0,0 +1,38 @@
+# Visible inner text
+
+(The "visible inner text" defined here is similar to, but not the same as, [visible text content][].)
+
+The <dfn>visible inner text of a node</dfn> depends on the kind of node. 
+
+The <dfn id="for-text">visible inner text of a [text node][]</dfn> is:
+-   if the [text node][] is [visible][], its visible inner text is its [data][];
+-   if the [text node][] is not-[visible][], [rendered][], and contains only [whitespace][], its visible inner text is the string `" "` (a single ASCII whitespace);
```

Comment:
The conditional here sounds a bit weird 🤔 
Notably, a text node that is not visible, rendered, and contains more than whitespace (e.g. in `<span style="visibility: hidden">Hello</span>`) would not trigger it and therefore have an empty string as visible inner text (rather than a whitespace).

Reply:
Interesting question.  I don't know the answer.  But I'll note that I copied this definition from [sanshikan ](https://github.com/Siteimprove/sanshikan/blob/main/terms/visible-inner-text.md) so if it needs fixing here, it probably needs fixing there too.

Reply:
OK, doing some archaeology, this is due to the fact that whitespace are not visible per our definition…
```html
<button aria-label="hello world"><span>hello</span><span id="space"> </span><span>world</span></button>
```
The `span#space` is not visible (and neither is its child text node). So the first bullet doesn't apply. Without the second bullet, the visible inner text of the button would be `helloworld`, not matching the accessible name of `hello world` due to spacing…
I guess we need to add an example to show that.

Reply:
This raises another question: what should we do with this?
`<a aria-label="Download specification" href="#"><span>Download</span><span style="visibility: hidden">x</span><span>specification</span></a>`
According to the current definition, because of the clause "contains only [whitespace][]", the visible inner text of the &lt;a&gt; element is "Downloadspecification".  Visually it looks like "Download specification".   So I wonder if we could remove the clause "contains only [whitespace][]".   What do you think?

Reply:
Good point 🤔 But if the `span` was invisible due to absolute positioning out of viewport, it shrould be removed:
```html
<a aria-label="Download specification" href="#"><span>Download</span><span style="position: absolute; left: -9999px">x</span><span>specification</span></a>
```

I guess the true condition is whether it creates a CSS box that lies somewhere between the ones of the rest of the text taking part in the computation (and isn't fully contained in them), or something like that 🙈 
Or maybe we just make the special case for `visibility: hidden` and assume that these is already a corner case and that it won't create too many true problems (We've been [using that definition](https://github.com/Siteimprove/alfa/blob/main/packages/alfa-rules/src/sia-r14/rule.ts#L97-L100) in Alfa [for two years](https://github.com/Siteimprove/alfa/pull/938) and I don't remember seeing a problem caused by it, so it may be safe to assume that it is a good enough approximation).

Reply:
This has given me a lot to think about.  I'll try to bring it up in our next one-on-one meeting.